xarray/tests/test_dataset.py

```diff
@@ -6698,7 +6698,7 @@ def test_polyfit_coord(self) -> None:
 
         out = ds.polyfit("numbers", 2, full=False)
         assert "var3_polyfit_coefficients" in out
-        assert "dim1" in out
+        assert "dim1" in out.dims
```

Comment:
this test was adding `dim1` as a *new* coordinate var (it is just a dimension name in `ds`). I *think* that's wrong.

cc @aulemahal

Reply:
Indeed, that seems better!